pkg/clustermanager/cluster_manager.go

```diff
@@ -341,12 +341,17 @@ func (c *ClusterManager) DeleteCluster(ctx context.Context, managementCluster, c
 }
 
 func (c *ClusterManager) UpgradeCluster(ctx context.Context, managementCluster, workloadCluster *types.Cluster, newClusterSpec *cluster.Spec, provider providers.Provider) error {
-	currentSpec, err := c.GetCurrentClusterSpec(ctx, workloadCluster, newClusterSpec.Cluster.Name)
+	eksaMgmtCluster := workloadCluster
+	if managementCluster != nil && managementCluster.ExistingManagement {
```

Comment:
Can we add a unit test for this scenario?

Reply:
Added